hphp/runtime/vm/jit/vasm-util.cpp

```diff
@@ -29,7 +29,12 @@ bool is_nop(const copy& i) { return i.s == i.d; }
 bool is_nop(const copy2& i) { return i.s0 == i.d0 && i.s1 == i.d1; }
 
 // movb r,r is a nop, however movl is not since it zeros upper bits.
-bool is_nop(const movb& i) { return i.s == i.d; }
+bool is_nop(const movb& i) {
+  // On PPC64 movb r,r cannot be considered a trivial nop since it zeros
```

Comment:
The problem is ppc64 doesn't move 8 bits like x86_64. The destiny is always a 64 bit register. A `movb` is translated to a ppc64 `rlwinm` instruction who rotates and mask and it's the proper way to move 8 (or more) bits between registers on POWER. The problem is we use `movb` on some lowered vasms like `cmpbm`, `xorb`, `andb` and we expect that `movb` will not be removed even if r(dst) = r(src). We use `movb` on `mozbl` and `movzbq` as we always zero upper bits on `movb` and the destiny is always a 64 bits register anyway. Unfortunately HHVM doesn't have a `movzb` vasm i think that not consider a `movb` a trivial nop on ppc64 is easiest than create a new vasm `movzb`. What do you think? should i create a new vasm `movzb` or you have another suggestion?